llvm/lib/CodeGen/InterleavedAccessPass.cpp

```diff
@@ -339,25 +361,45 @@ bool InterleavedAccessImpl::lowerInterleavedLoad(
     return false;
 
   bool BinOpShuffleChanged =
-      replaceBinOpShuffles(BinOpShuffles.getArrayRef(), Shuffles, LI);
+      replaceBinOpShuffles(BinOpShuffles.getArrayRef(), Shuffles, Load);
 
-  LLVM_DEBUG(dbgs() << "IA: Found an interleaved load: " << *LI << "\n");
+  if (auto *VPLoad = dyn_cast<VPIntrinsic>(Load)) {
+    Value *LaneMask = getMask(VPLoad->getArgOperand(1), Factor,
+                              cast<VectorType>(Shuffles[0]->getType()));
```

Comment:
```suggestion
                              VecTy);
```

Reply:
Fixed. Though it should be `cast<VectorType>(VecTy)`